File: src/main/java/core/procedures/strings/Trim.java
package core.procedures.strings;

import core.procedures.AFn;
import core.procedures.FnArgsBuilder;

public final class Trim extends AFn {

  public Trim() {
    super(new FnArgsBuilder().min(1).max(1).mandatory(new Class[]{CharSequence.class}).build());
  }

  @Override
  public String getName() {
    return "trim";
  }

  @Override
  public String apply1(Object arg) {
    return arg.toString().trim();
  }
}

File: src/main/kotlin/core/procedures/strings/Trim.kt
package core.procedures.strings

import core.procedures.AFn

class Trim : AFn<CharSequence?, String>(name = "trim", isPure = true, minArgs = 1, maxArgs = 1,
                 mandatoryArgsTypes = arrayOf<Class<*>>(CharSequence::class.java)) {

    override operator fun invoke(arg: CharSequence?) = arg!!.toString().trim()
}
